### DISCUSSION TITLE
Generate an interface with the generated methods?

### DISCUSSION THREAD

**Discussion by pengux** [sqlc-dev/sqlc#2831] [Q&A] [answered]
### What do you want to change?

Create an interface for `Queries` (perhaps named `Dber`) which contains all generated methods. This will allow users to mock the interface in tests and have the interface be updated automatically whenever the queries are updated.

### What programming language backends need to be changed?

Go

**Comment by kyleconroy** [comment, marked as answer]
The `emit_interface` option does exactly what you want.

```json
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql",
      "emit_interface": true
    }
  ]
}
```

https://play.sqlc.dev/p/1f0fc5978c9788beadd198a562fea3f3ed073dc3b4eaca63cfe9c84eb7655e20

**Reply by pengux** [reply]
Ah, not sure how I missed that in the docs, thank you!
